### PR TITLE
Renaming TestPrecondition utility methods

### PR DESCRIPTION
**Pull request by hegyibalint**
The utility methods in `TestPrecondition` had inconsistent naming, and some were flat-out grammatically incorrect.
This PR fixes that issue.

### CODE REVIEW

**Comment by hegyibalint** [comment]
Renaming happened here:
[subprojects/internal-testing/src/main/groovy/org/gradle/test/precondition/TestPrecondition.java](https://github.com/gradle/gradle/pull/25280/files#diff-049c6839099bcca1b9f7a92ee0e1f8d5ab2e265278db4606c1877d3cf87d005f)

Everything else should be usage renaming.

**Review by lptr — APPROVED** [review]

File: subprojects/internal-testing/src/main/groovy/org/gradle/test/preconditions/UnitTestPreconditions.groovy [hunk 1]
```diff
@@ -22,51 +22,48 @@ import org.gradle.internal.os.OperatingSystem
 import org.gradle.test.precondition.TestPrecondition
 import org.testcontainers.DockerClientFactory
 
-import static org.gradle.test.precondition.TestPrecondition.doSatisfies;
```
Comment (wolfs) [inline subprojects/internal-testing/src/main/groovy/org/gradle/test/preconditions/UnitTestPreconditions.groovy:25]: The static import is missing and that is why compilation fails.

Comment (hegyibalint) [reply subprojects/internal-testing/src/main/groovy/org/gradle/test/preconditions/UnitTestPreconditions.groovy:25]: For some reason, IntelliJ believes that these static imports are not needed...

File: subprojects/internal-testing/src/main/groovy/org/gradle/test/precondition/TestPrecondition.java [hunk 2]
```diff
@@ -26,28 +26,51 @@
...
         return precondition.isSatisfied();
     }
 
-    static boolean doSatisfiesAll(Class<? extends TestPrecondition>[] preconditionClasses) throws Exception {
+    /**
+     * Utility method to check if all precondition classes are satisfied.
+     *
+     * @param preconditionClasses a list of precondition classes to be checked
+     * @return true if all preconditions are satisfied
+     * @throws Exception if a precondition cannot be checked
```
Comment (wolfs) [inline subprojects/internal-testing/src/main/groovy/org/gradle/test/precondition/TestPrecondition.java:56]: 💄 I do find the `@param`, `@return` and `@throws` somewhat superfluous, though I also don't feel strongly in removing them.

**Review by wolfs — APPROVED** [review]
Much better. Please fix the compile errors.

**Comment by hegyibalint** [comment]
@bot-gradle test and merge please

**Comment by hegyibalint** [comment]
@bot-gradle test and merge

**Pull request merged**